### PR TITLE
Added log level option to commands.

### PR DESCRIPTION
**Pull request by patriksvensson**
This is a proposal for fixing #120.

I'm not very familiar with your code base, so I apologize if it there are discrepancies in code style or similar. I've tried to keep the code as idiomatic to the one in the code base.

If you have any feedback for changes and/or improvements I will happily change my code. : smile:

## ApiCommand based commands

image

## Pack command

image

### CODE REVIEW

**Comment by mcasperson** [comment]
Hi Patrik, thanks for this PR. We like the idea, but unfortunately in its current for the new logging options don't work as expected.

In Program.cs, this code:

```
                if (command is ILogLevelAware logLevelAwareCommand)
                {
                    if (logLevelAwareCommand.LogLevel != null)
                    {
                        ConfigureLogger(logLevelAwareCommand.LogLevel);
                    }
                }
```

is executed before this code:

```
command.Execute(args.Skip(1).ToArray()).GetAwaiter().GetResult();
```

`command.Execute` is where the command line params are parsed. So prior to `command.Execute`, `logLevelAwareCommand.LogLevel` is always null. This means the log level is never changed.

**Comment by patriksvensson** [comment]
@mcasperson Ok. That’s strange. I will take a look at it but I tried it out and it seemed to work ad expected...

I will take a look and update the PR in a couple hours.

**Comment by patriksvensson** [comment]
@mcaspwrson Ok, I see it now. Seems to be an issue of ”10 unit tests, 0 integration tests” :smile:

The fix should be easy. I’ll update te PR in a couple of hours.

**Comment by patriksvensson** [comment]
@mcasperson Ok, I've updated the PR. It's not very testable in the current state, but I knowingly opted for a simple solution which didn't require us to inject Serilog's `LoggingLevelSwitch` in every command and we can take the discussion from there.

Another approach we could take is to create a decorator for `ILogger` that contains the logging level switch directly. This would make the log level code more testable, but also increase the complexity + maintainability of the code base.

It's possible I've missed something obvious here, if that's the case, please don't hesitate to correct me :smile:

File: source/Octo/Diagnostics/LogUtilities.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            LevelSwitch = new LoggingLevelSwitch(LogEventLevel.Debug);
+        }
+        
+        public static LogEventLevel ParseLogLevel(string value)
+        {
+            if (_lookup.TryGetValue(value, out var level))
+            {
+                return level;
+            }
+            throw new CommandException($"Unrecognized loglevel '{value}'.");
```
Comment (mcasperson) [inline source/Octo/Diagnostics/LogUtilities.cs]: We try to provide valid options when displaying an error. So this code would print:

> Unrecognized loglevel 'blah'.

Ideally it should print something like:

> Unrecognized loglevel 'blah'. Valid options are fatal, error, warning, information, debug or verbose.

**Comment by patriksvensson** [comment]
@mcasperson I've updated the message per your request and I also did some minor changes to the description of the command to match other commands better.

I've also rebased my branch against master.

**Comment by mcasperson** [comment]
Thanks for that Patrik. We've merged this, and the changes will go out with the next release.

**Comment by patriksvensson** [comment]
@mcasperson Perfect! Do you have any estimation of when you're going to push the new client?

**Pull request merged**